### PR TITLE
Convert types to TType convention

### PR DESCRIPTION
**Pull request by bhourahine**
Also changed two colliding logical variables from tGeoOpt to
isGeoOpt and tLinResp to isLinResp

### CODE REVIEW

File: prog/dftb+/lib_md/berendsentherm.F90 [hunk 1]
```diff
@@ -21,32 +21,32 @@ module dftbp_berendsentherm
...
-    type(OTempProfile), pointer :: pTempProfile
+    type(tTempProfile), pointer :: pTempProfile
 
     !> coupling strength to friction term
     real(dp) :: couplingParameter
 
     !> MD Framework.
-    type(OMDCommon) :: pMDFrame
-  end type OBerendsenThermostat
+    type(tMDCommon) :: pMDFrame
```
Comment (awvwgk) [inline prog/dftb+/lib_md/berendsentherm.F90]: `TMDCommon`?

File: prog/dftb+/lib_md/xlbomd.F90 [hunk 2]
```diff
@@ -19,15 +19,15 @@ module dftbp_xlbomd
   implicit none
   private
 
-  public :: XlbomdInp, Xlbomd, Xlbomd_init
+  public :: TXlbomdInp, TXlbomd, Xlbomd_init
```
Comment (awvwgk) [inline prog/dftb+/lib_md/xlbomd.F90]: Is it now `TXlBOMD` or `TXlbomd`? Was also not consistent before.

File: prog/misc/slakovalue/splvalue.F90 [hunk 3]
```diff
@@ -19,7 +19,7 @@ program splvalue
   character(*), parameter :: fname = "test.skf"
   character(lc) :: arg
   type(trepsplinein) :: repsplinein
-  type(orepspline) :: prepspline
+  type(trepspline) :: prepspline
```
Comment (awvwgk) [inline prog/misc/slakovalue/splvalue.F90]: `TRepSpline`?

File: prog/waveplot/gridcache.F90 [hunk 4]
```diff
@@ -25,10 +25,10 @@ module dftbp_gridcache
 
 
   !> Contains the data for a grid cache
-  type OGridCache
+  type TgridCache
 
     !> Molecular orbital calculator
-    type(OMolecularOrbital), pointer :: molorb
+    type(tMolecularOrbital), pointer :: molorb
```
Comment (awvwgk) [inline prog/waveplot/gridcache.F90]: `TMolecularOrbital`?

File: prog/waveplot/gridcache.F90 [hunk 5]
```diff
@@ -165,7 +165,7 @@ subroutine GridCache_init(sf, levelIndex, nOrb, nAllLevel, &
     logical, intent(in) :: tReal
 
     !> Molecular orbital calculator
-    type(OMolecularOrbital), pointer, intent(in) :: molorb
+    type(tMolecularOrbital), pointer, intent(in) :: molorb
```
Comment (awvwgk) [inline prog/waveplot/gridcache.F90]: `TMolecularOrbital`?

File: prog/dftb+/lib_dftbplus/inputdata.F90 [hunk 6]
```diff
@@ -454,19 +464,21 @@ module dftbp_inputdata
 
 
     !> XLBOMD
-    type(XLBOMDInp), allocatable :: xlbomd
+    type(TXLBOMDInp), allocatable :: xlbomd
```
[the remark this answers is not in the document]
Comment (bhourahine) [reply prog/dftb+/lib_dftbplus/inputdata.F90]: See commit comment

**Pull request merged**